{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "4uw9Xz6AzNvn"
      },
      "outputs": [],
      "source": [
        "%load_ext autoreload\n",
        "%autoreload 2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Introduction"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "llOsx7rYHki1"
      },
      "source": [
        "## Setup"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: genomic-benchmarks in ./venv/lib/python3.8/site-packages (0.0.6)\n",
            "Requirement already satisfied: pip>=20.0.1 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (21.3.1)\n",
            "Requirement already satisfied: requests>=2.23.0 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (2.27.1)\n",
            "Requirement already satisfied: yarl in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (1.7.2)\n",
            "Requirement already satisfied: pandas>=1.1.4 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (1.4.0)\n",
            "Requirement already satisfied: pyyaml>=5.3.1 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (6.0)\n",
            "Requirement already satisfied: googledrivedownloader>=0.4 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (0.4)\n",
            "Requirement already satisfied: tqdm>=4.41.1 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (4.62.3)\n",
            "Requirement already satisfied: numpy>=1.17.0 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (1.22.2)\n",
            "Requirement already satisfied: biopython>=1.79 in ./venv/lib/python3.8/site-packages (from genomic-benchmarks) (1.79)\n",
            "Requirement already satisfied: pytz>=2020.1 in ./venv/lib/python3.8/site-packages (from pandas>=1.1.4->genomic-benchmarks) (2021.3)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in ./venv/lib/python3.8/site-packages (from pandas>=1.1.4->genomic-benchmarks) (2.8.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in ./venv/lib/python3.8/site-packages (from requests>=2.23.0->genomic-benchmarks) (3.3)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./venv/lib/python3.8/site-packages (from requests>=2.23.0->genomic-benchmarks) (1.26.8)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib/python3.8/site-packages (from requests>=2.23.0->genomic-benchmarks) (2021.10.8)\n",
            "Requirement already satisfied: charset-normalizer~=2.0.0 in ./venv/lib/python3.8/site-packages (from requests>=2.23.0->genomic-benchmarks) (2.0.11)\n",
            "Requirement already satisfied: multidict>=4.0 in ./venv/lib/python3.8/site-packages (from yarl->genomic-benchmarks) (6.0.2)\n",
            "Requirement already satisfied: six>=1.5 in ./venv/lib/python3.8/site-packages (from python-dateutil>=2.8.1->pandas>=1.1.4->genomic-benchmarks) (1.16.0)\n",
            "\u001b[33mWARNING: You are using pip version 21.3.1; however, version 22.0.4 is available.\n",
            "You should consider upgrading via the '/home/katarina/git/rbp_encode_eclip/venv/bin/python -m pip install --upgrade pip' command.\u001b[0m\n",
            "Requirement already satisfied: sklearn in ./venv/lib/python3.8/site-packages (0.0)\n",
            "Requirement already satisfied: scikit-learn in ./venv/lib/python3.8/site-packages (from sklearn) (1.0.2)\n",
            "Requirement already satisfied: threadpoolctl>=2.0.0 in ./venv/lib/python3.8/site-packages (from scikit-learn->sklearn) (3.1.0)\n",
            "Requirement already satisfied: numpy>=1.14.6 in ./venv/lib/python3.8/site-packages (from scikit-learn->sklearn) (1.22.2)\n",
            "Requirement already satisfied: joblib>=0.11 in ./venv/lib/python3.8/site-packages (from scikit-learn->sklearn) (1.1.0)\n",
            "Requirement already satisfied: scipy>=1.1.0 in ./venv/lib/python3.8/site-packages (from scikit-learn->sklearn) (1.8.0)\n",
            "\u001b[33mWARNING: You are using pip version 21.3.1; however, version 22.0.4 is available.\n",
            "You should consider upgrading via the '/home/katarina/git/rbp_encode_eclip/venv/bin/python -m pip install --upgrade pip' command.\u001b[0m\n",
            "Requirement already satisfied: ipywidgets in ./venv/lib/python3.8/site-packages (7.6.5)\n",
            "Requirement already satisfied: ipython>=4.0.0 in ./venv/lib/python3.8/site-packages (from ipywidgets) (8.0.1)\n",
            "Requirement already satisfied: ipython-genutils~=0.2.0 in ./venv/lib/python3.8/site-packages (from ipywidgets) (0.2.0)\n",
            "Requirement already satisfied: nbformat>=4.2.0 in ./venv/lib/python3.8/site-packages (from ipywidgets) (5.1.3)\n",
            "Requirement already satisfied: ipykernel>=4.5.1 in ./venv/lib/python3.8/site-packages (from ipywidgets) (6.9.0)\n",
            "Requirement already satisfied: traitlets>=4.3.1 in ./venv/lib/python3.8/site-packages (from ipywidgets) (5.1.1)\n",
            "Requirement already satisfied: widgetsnbextension~=3.5.0 in ./venv/lib/python3.8/site-packages (from ipywidgets) (3.5.2)\n",
            "Requirement already satisfied: jupyterlab-widgets>=1.0.0 in ./venv/lib/python3.8/site-packages (from ipywidgets) (1.0.2)\n",
            "Requirement already satisfied: jupyter-client<8.0 in ./venv/lib/python3.8/site-packages (from ipykernel>=4.5.1->ipywidgets) (7.1.2)\n",
            "Requirement already satisfied: debugpy<2.0,>=1.0.0 in ./venv/lib/python3.8/site-packages (from ipykernel>=4.5.1->ipywidgets) (1.5.1)\n",
            "Requirement already satisfied: nest-asyncio in ./venv/lib/python3.8/site-packages (from ipykernel>=4.5.1->ipywidgets) (1.5.4)\n",
            "Requirement already satisfied: matplotlib-inline<0.2.0,>=0.1.0 in ./venv/lib/python3.8/site-packages (from ipykernel>=4.5.1->ipywidgets) (0.1.3)\n",
            "Requirement already satisfied: tornado<7.0,>=4.2 in ./venv/lib/python3.8/site-packages (from ipykernel>=4.5.1->ipywidgets) (6.1)\n",
            "Requirement already satisfied: decorator in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (5.1.1)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (3.0.27)\n",
            "Requirement already satisfied: setuptools>=18.5 in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (60.5.0)\n",
            "Requirement already satisfied: pickleshare in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (0.7.5)\n",
            "Requirement already satisfied: stack-data in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (0.1.4)\n",
            "Requirement already satisfied: pexpect>4.3 in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (4.8.0)\n",
            "Requirement already satisfied: jedi>=0.16 in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (0.18.1)\n",
            "Requirement already satisfied: backcall in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (0.2.0)\n",
            "Requirement already satisfied: pygments in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (2.11.2)\n",
            "Requirement already satisfied: black in ./venv/lib/python3.8/site-packages (from ipython>=4.0.0->ipywidgets) (22.1.0)\n",
            "Requirement already satisfied: jupyter-core in ./venv/lib/python3.8/site-packages (from nbformat>=4.2.0->ipywidgets) (4.9.1)\n",
            "Requirement already satisfied: jsonschema!=2.5.0,>=2.4 in ./venv/lib/python3.8/site-packages (from nbformat>=4.2.0->ipywidgets) (4.4.0)\n",
            "Requirement already satisfied: notebook>=4.4.1 in ./venv/lib/python3.8/site-packages (from widgetsnbextension~=3.5.0->ipywidgets) (6.4.8)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.0 in ./venv/lib/python3.8/site-packages (from jedi>=0.16->ipython>=4.0.0->ipywidgets) (0.8.3)\n",
            "Requirement already satisfied: importlib-resources>=1.4.0 in ./venv/lib/python3.8/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets) (5.4.0)\n",
            "Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in ./venv/lib/python3.8/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets) (0.18.1)\n",
            "Requirement already satisfied: attrs>=17.4.0 in ./venv/lib/python3.8/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets) (21.4.0)\n",
            "Requirement already satisfied: pyzmq>=13 in ./venv/lib/python3.8/site-packages (from jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (22.3.0)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in ./venv/lib/python3.8/site-packages (from jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (2.8.2)\n",
            "Requirement already satisfied: entrypoints in ./venv/lib/python3.8/site-packages (from jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (0.4)\n",
            "Requirement already satisfied: argon2-cffi in ./venv/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (21.3.0)\n",
            "Requirement already satisfied: Send2Trash>=1.8.0 in ./venv/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (1.8.0)\n",
            "Requirement already satisfied: prometheus-client in ./venv/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.13.1)\n",
            "Requirement already satisfied: terminado>=0.8.3 in ./venv/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.13.1)\n",
            "Requirement already satisfied: nbconvert in ./venv/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (6.4.2)\n",
            "Requirement already satisfied: jinja2 in ./venv/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (3.0.3)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in ./venv/lib/python3.8/site-packages (from pexpect>4.3->ipython>=4.0.0->ipywidgets) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in ./venv/lib/python3.8/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=4.0.0->ipywidgets) (0.2.5)\n",
            "Requirement already satisfied: click>=8.0.0 in ./venv/lib/python3.8/site-packages (from black->ipython>=4.0.0->ipywidgets) (8.0.3)\n",
            "Requirement already satisfied: tomli>=1.1.0 in ./venv/lib/python3.8/site-packages (from black->ipython>=4.0.0->ipywidgets) (2.0.1)\n",
            "Requirement already satisfied: typing-extensions>=3.10.0.0 in ./venv/lib/python3.8/site-packages (from black->ipython>=4.0.0->ipywidgets) (4.0.1)\n",
            "Requirement already satisfied: mypy-extensions>=0.4.3 in ./venv/lib/python3.8/site-packages (from black->ipython>=4.0.0->ipywidgets) (0.4.3)\n",
            "Requirement already satisfied: platformdirs>=2 in ./venv/lib/python3.8/site-packages (from black->ipython>=4.0.0->ipywidgets) (2.5.0)\n",
            "Requirement already satisfied: pathspec>=0.9.0 in ./venv/lib/python3.8/site-packages (from black->ipython>=4.0.0->ipywidgets) (0.9.0)\n",
            "Requirement already satisfied: executing in ./venv/lib/python3.8/site-packages (from stack-data->ipython>=4.0.0->ipywidgets) (0.8.2)\n",
            "Requirement already satisfied: asttokens in ./venv/lib/python3.8/site-packages (from stack-data->ipython>=4.0.0->ipywidgets) (2.0.5)\n",
            "Requirement already satisfied: pure-eval in ./venv/lib/python3.8/site-packages (from stack-data->ipython>=4.0.0->ipywidgets) (0.2.2)\n",
            "Requirement already satisfied: zipp>=3.1.0 in ./venv/lib/python3.8/site-packages (from importlib-resources>=1.4.0->jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets) (3.7.0)\n",
            "Requirement already satisfied: six>=1.5 in ./venv/lib/python3.8/site-packages (from python-dateutil>=2.1->jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (1.16.0)\n",
            "Requirement already satisfied: argon2-cffi-bindings in ./venv/lib/python3.8/site-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (21.2.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in ./venv/lib/python3.8/site-packages (from jinja2->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (2.0.1)\n",
            "Requirement already satisfied: defusedxml in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.7.1)\n",
            "Requirement already satisfied: testpath in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.0)\n",
            "Requirement already satisfied: mistune<2,>=0.8.1 in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.8.4)\n",
            "Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.10)\n",
            "Requirement already satisfied: jupyterlab-pygments in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.1.2)\n",
            "Requirement already satisfied: pandocfilters>=1.4.1 in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (1.5.0)\n",
            "Requirement already satisfied: bleach in ./venv/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (4.1.0)\n",
            "Requirement already satisfied: cffi>=1.0.1 in ./venv/lib/python3.8/site-packages (from argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (1.15.0)\n",
            "Requirement already satisfied: webencodings in ./venv/lib/python3.8/site-packages (from bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.1)\n",
            "Requirement already satisfied: packaging in ./venv/lib/python3.8/site-packages (from bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (21.3)\n",
            "Requirement already satisfied: pycparser in ./venv/lib/python3.8/site-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (2.21)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in ./venv/lib/python3.8/site-packages (from packaging->bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (3.0.7)\n",
            "\u001b[33mWARNING: You are using pip version 21.3.1; however, version 22.0.4 is available.\n",
            "You should consider upgrading via the '/home/katarina/git/rbp_encode_eclip/venv/bin/python -m pip install --upgrade pip' command.\u001b[0m\n",
            "Enabling notebook extension jupyter-js-widgets/extension...\n",
            "      - Validating: \u001b[32mOK\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!pip install sklearn\n",
        "!pip install ipywidgets\n",
        "!jupyter nbextension enable --py widgetsnbextension"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "GAzDXv4t1xdF"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "from urllib import request\n",
        "from tqdm import tqdm_notebook\n",
        "from pathlib import Path\n",
        "import yaml"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cDj4TAXvH5yu"
      },
      "source": [
        "## Download metadata file from Encode"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Qb4mZtuezTI3",
        "outputId": "4a40792d-2dcc-46e7-f9e9-9edcfdef9300"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2022-04-09 12:54:58--  https://www.encodeproject.org/metadata/?status=released&internal_tags=ENCORE&assay_title=eCLIP&biosample_ontology.term_name=K562&biosample_ontology.term_name=HepG2&files.file_type=bed+narrowPeak&type=Experiment&files.analyses.status=released&files.preferred_default=true\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: unspecified [text/tsv]\n",
            "Saving to: ‘metadata.tsv’\n",
            "\n",
            "metadata.tsv            [   <=>              ] 125,40K   153KB/s    in 0,8s    \n",
            "\n",
            "2022-04-09 12:55:00 (153 KB/s) - ‘metadata.tsv’ saved [128408]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget -O metadata.tsv \"https://www.encodeproject.org/metadata/?status=released&internal_tags=ENCORE&assay_title=eCLIP&biosample_ontology.term_name=K562&biosample_ontology.term_name=HepG2&files.file_type=bed+narrowPeak&type=Experiment&files.analyses.status=released&files.preferred_default=true\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "pc8w9GAv1zcg"
      },
      "outputs": [],
      "source": [
        "metadata = pd.read_csv('metadata.tsv', sep='\\t')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 860
        },
        "id": "_R1faHmV17Vn",
        "outputId": "e8a3a713-63b6-4329-c1b1-8e8ef9cb8865"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>File accession</th>\n",
              "      <th>File format</th>\n",
              "      <th>File type</th>\n",
              "      <th>File format type</th>\n",
              "      <th>Output type</th>\n",
              "      <th>File assembly</th>\n",
              "      <th>Experiment accession</th>\n",
              "      <th>Assay</th>\n",
              "      <th>Donor(s)</th>\n",
              "      <th>Biosample term id</th>\n",
              "      <th>...</th>\n",
              "      <th>Genome annotation</th>\n",
              "      <th>Platform</th>\n",
              "      <th>Controlled by</th>\n",
              "      <th>File Status</th>\n",
              "      <th>s3_uri</th>\n",
              "      <th>File analysis title</th>\n",
              "      <th>File analysis status</th>\n",
              "      <th>Audit WARNING</th>\n",
              "      <th>Audit NOT_COMPLIANT</th>\n",
              "      <th>Audit ERROR</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ENCFF311PHS</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR735HOK</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAC/</td>\n",
              "      <td>EFO:0001187</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/dca28e48-1652-4c...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>antibody characterized with exemption</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ENCFF203AWR</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR356MSW</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAD/</td>\n",
              "      <td>EFO:0002067</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/2a02265d-be38-42...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>mixed read lengths</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ENCFF890DJI</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR820UYE</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAC/</td>\n",
              "      <td>EFO:0001187</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/b217e442-31c4-47...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ENCFF484NSW</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR734ZHL</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAD/</td>\n",
              "      <td>EFO:0002067</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/2bd059f7-b792-4e...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ENCFF093FTO</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR202HKN</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAD/</td>\n",
              "      <td>EFO:0002067</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/d1fb8b83-7d6a-45...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>218</th>\n",
              "      <td>ENCFF886MOW</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR893NWB</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAC/</td>\n",
              "      <td>EFO:0001187</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2016/11/30/ffb7042d-9913-41...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>219</th>\n",
              "      <td>ENCFF507ZJR</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR989SMC</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAD/</td>\n",
              "      <td>EFO:0002067</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2016/11/30/321fb00b-284d-40...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>mixed read lengths</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>220</th>\n",
              "      <td>ENCFF100OEX</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR384KAN</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAC/</td>\n",
              "      <td>EFO:0001187</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2016/11/30/633ef9cd-a5d2-43...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>221</th>\n",
              "      <td>ENCFF882APR</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR977OXG</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAC/</td>\n",
              "      <td>EFO:0001187</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/3cdea817-21b9-4d...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>222</th>\n",
              "      <td>ENCFF766DUS</td>\n",
              "      <td>bed narrowPeak</td>\n",
              "      <td>bed</td>\n",
              "      <td>narrowPeak</td>\n",
              "      <td>peaks</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>ENCSR922WJV</td>\n",
              "      <td>eCLIP</td>\n",
              "      <td>/human-donors/ENCDO000AAD/</td>\n",
              "      <td>EFO:0002067</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>released</td>\n",
              "      <td>s3://encode-public/2018/04/29/a7e56a50-faf8-4a...</td>\n",
              "      <td>Lab custom GRCh38</td>\n",
              "      <td>released</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>223 rows × 58 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "    File accession     File format File type File format type Output type  \\\n",
              "0      ENCFF311PHS  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "1      ENCFF203AWR  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "2      ENCFF890DJI  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "3      ENCFF484NSW  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "4      ENCFF093FTO  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "..             ...             ...       ...              ...         ...   \n",
              "218    ENCFF886MOW  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "219    ENCFF507ZJR  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "220    ENCFF100OEX  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "221    ENCFF882APR  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "222    ENCFF766DUS  bed narrowPeak       bed       narrowPeak       peaks   \n",
              "\n",
              "    File assembly Experiment accession  Assay                    Donor(s)  \\\n",
              "0          GRCh38          ENCSR735HOK  eCLIP  /human-donors/ENCDO000AAC/   \n",
              "1          GRCh38          ENCSR356MSW  eCLIP  /human-donors/ENCDO000AAD/   \n",
              "2          GRCh38          ENCSR820UYE  eCLIP  /human-donors/ENCDO000AAC/   \n",
              "3          GRCh38          ENCSR734ZHL  eCLIP  /human-donors/ENCDO000AAD/   \n",
              "4          GRCh38          ENCSR202HKN  eCLIP  /human-donors/ENCDO000AAD/   \n",
              "..            ...                  ...    ...                         ...   \n",
              "218        GRCh38          ENCSR893NWB  eCLIP  /human-donors/ENCDO000AAC/   \n",
              "219        GRCh38          ENCSR989SMC  eCLIP  /human-donors/ENCDO000AAD/   \n",
              "220        GRCh38          ENCSR384KAN  eCLIP  /human-donors/ENCDO000AAC/   \n",
              "221        GRCh38          ENCSR977OXG  eCLIP  /human-donors/ENCDO000AAC/   \n",
              "222        GRCh38          ENCSR922WJV  eCLIP  /human-donors/ENCDO000AAD/   \n",
              "\n",
              "    Biosample term id  ... Genome annotation Platform Controlled by  \\\n",
              "0         EFO:0001187  ...               NaN      NaN           NaN   \n",
              "1         EFO:0002067  ...               NaN      NaN           NaN   \n",
              "2         EFO:0001187  ...               NaN      NaN           NaN   \n",
              "3         EFO:0002067  ...               NaN      NaN           NaN   \n",
              "4         EFO:0002067  ...               NaN      NaN           NaN   \n",
              "..                ...  ...               ...      ...           ...   \n",
              "218       EFO:0001187  ...               NaN      NaN           NaN   \n",
              "219       EFO:0002067  ...               NaN      NaN           NaN   \n",
              "220       EFO:0001187  ...               NaN      NaN           NaN   \n",
              "221       EFO:0001187  ...               NaN      NaN           NaN   \n",
              "222       EFO:0002067  ...               NaN      NaN           NaN   \n",
              "\n",
              "     File Status                                             s3_uri  \\\n",
              "0       released  s3://encode-public/2018/04/29/dca28e48-1652-4c...   \n",
              "1       released  s3://encode-public/2018/04/29/2a02265d-be38-42...   \n",
              "2       released  s3://encode-public/2018/04/29/b217e442-31c4-47...   \n",
              "3       released  s3://encode-public/2018/04/29/2bd059f7-b792-4e...   \n",
              "4       released  s3://encode-public/2018/04/29/d1fb8b83-7d6a-45...   \n",
              "..           ...                                                ...   \n",
              "218     released  s3://encode-public/2016/11/30/ffb7042d-9913-41...   \n",
              "219     released  s3://encode-public/2016/11/30/321fb00b-284d-40...   \n",
              "220     released  s3://encode-public/2016/11/30/633ef9cd-a5d2-43...   \n",
              "221     released  s3://encode-public/2018/04/29/3cdea817-21b9-4d...   \n",
              "222     released  s3://encode-public/2018/04/29/a7e56a50-faf8-4a...   \n",
              "\n",
              "     File analysis title  File analysis status  \\\n",
              "0      Lab custom GRCh38              released   \n",
              "1      Lab custom GRCh38              released   \n",
              "2      Lab custom GRCh38              released   \n",
              "3      Lab custom GRCh38              released   \n",
              "4      Lab custom GRCh38              released   \n",
              "..                   ...                   ...   \n",
              "218    Lab custom GRCh38              released   \n",
              "219    Lab custom GRCh38              released   \n",
              "220    Lab custom GRCh38              released   \n",
              "221    Lab custom GRCh38              released   \n",
              "222    Lab custom GRCh38              released   \n",
              "\n",
              "                             Audit WARNING  Audit NOT_COMPLIANT  Audit ERROR  \n",
              "0    antibody characterized with exemption                  NaN          NaN  \n",
              "1                       mixed read lengths                  NaN          NaN  \n",
              "2                                      NaN                  NaN          NaN  \n",
              "3                                      NaN                  NaN          NaN  \n",
              "4                                      NaN                  NaN          NaN  \n",
              "..                                     ...                  ...          ...  \n",
              "218                                    NaN                  NaN          NaN  \n",
              "219                     mixed read lengths                  NaN          NaN  \n",
              "220                                    NaN                  NaN          NaN  \n",
              "221                                    NaN                  NaN          NaN  \n",
              "222                                    NaN                  NaN          NaN  \n",
              "\n",
              "[223 rows x 58 columns]"
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "metadata"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lCHm_-85D5rc"
      },
      "source": [
        "## Prepare directory structure"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "fBkbae9LB2lx"
      },
      "outputs": [],
      "source": [
        "BASE_FILE_PATH = Path(\"./csv/\")\n",
        "\n",
        "# copied from https://stackoverflow.com/a/57892171\n",
        "def rm_tree(pth: Path):\n",
        "    for child in pth.iterdir():\n",
        "        if child.is_file():\n",
        "            child.unlink()\n",
        "        else:\n",
        "            rm_tree(child)\n",
        "    pth.rmdir()\n",
        "\n",
        "if BASE_FILE_PATH.exists():\n",
        "    rm_tree(BASE_FILE_PATH)\n",
        "    \n",
        "BASE_FILE_PATH.mkdir()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [],
      "source": [
        "!rm metadata.tsv"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pWxOSvWNIJes"
      },
      "source": [
        "## Transform data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ea0hQEp32JYo",
        "outputId": "6f1a97d4-5efa-45a0-85f0-b67610efe100"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "array(['YBX3-human', 'SSB-human', 'PABPN1-human', 'UTP3-human',\n",
              "       'WDR3-human', 'NIPBL-human', 'WDR43-human', 'EXOSC5-human',\n",
              "       'UPF1-human', 'APOBEC3C-human', 'WRN-human', 'PABPC4-human',\n",
              "       'FUS-human', 'SUPV3L1-human', 'ZC3H8-human', 'PHF6-human',\n",
              "       'FXR2-human', 'FXR1-human', 'TROVE2-human', 'PCBP2-human',\n",
              "       'SUB1-human', 'SF3B1-human', 'RBFOX2-human', 'AGGF1-human',\n",
              "       'SDAD1-human', 'HNRNPL-human', 'HNRNPUL1-human', 'KHDRBS1-human',\n",
              "       'FUBP3-human', 'YWHAG-human', 'FASTKD2-human', 'SND1-human',\n",
              "       'NONO-human', 'TARDBP-human', 'DDX52-human', 'DDX51-human',\n",
              "       'STAU2-human', 'BCLAF1-human', 'HLTF-human', 'AQR-human',\n",
              "       'HNRNPC-human', 'AKAP8L-human', 'DHX30-human', 'AATF-human',\n",
              "       'DDX6-human', 'SBDS-human', 'LARP4-human', 'DDX55-human',\n",
              "       'NSUN2-human', 'U2AF2-human', 'METAP2-human', 'LSM11-human',\n",
              "       'DDX21-human', 'ABCF1-human', 'SLTM-human', 'SUGP2-human',\n",
              "       'TIAL1-human', 'SMNDC1-human', 'BUD13-human', 'UCHL5-human',\n",
              "       'SAFB2-human', 'NKRF-human', 'EIF3D-human', 'ZC3H11A-human',\n",
              "       'SRSF1-human', 'XRCC6-human', 'HNRNPK-human', 'RBM15-human',\n",
              "       'RPS3-human', 'SERBP1-human', 'UTP18-human', 'GTF2F1-human',\n",
              "       'GPKOW-human', 'DGCR8-human', 'U2AF1-human', 'PPIG-human',\n",
              "       'HNRNPU-human', 'PCBP1-human', 'TRA2A-human', 'NPM1-human',\n",
              "       'RBM5-human', 'SF3B4-human', 'TAF15-human', 'ZRANB2-human',\n",
              "       'DDX59-human', 'PUS1-human', 'LIN28B-human', 'NCBP2-human',\n",
              "       'POLR2G-human', 'GNL3-human', 'DKC1-human', 'PUM2-human',\n",
              "       'QKI-human', 'PUM1-human', 'G3BP1-human', 'NIP7-human',\n",
              "       'SAFB-human', 'ZNF800-human', 'EFTUD2-human', 'DROSHA-human',\n",
              "       'EIF3H-human', 'NOLC1-human', 'BCCIP-human', 'EWSR1-human',\n",
              "       'IGF2BP1-human', 'CSTF2T-human', 'TIA1-human', 'FKBP4-human',\n",
              "       'LARP7-human', 'AARS-human', 'HNRNPM-human', 'FTO-human',\n",
              "       'SRSF7-human', 'MATR3-human', 'ILF3-human', 'FAM120A-human',\n",
              "       'SLBP-human', 'RBM22-human', 'DDX42-human', 'IGF2BP2-human',\n",
              "       'HNRNPA1-human', 'KHSRP-human', 'CSTF2-human', 'XPO5-human',\n",
              "       'IGF2BP3-human', 'ZNF622-human', 'SF3A3-human', 'GRWD1-human',\n",
              "       'SFPQ-human', 'AKAP1-human', 'PRPF8-human', 'CDC40-human',\n",
              "       'DDX3X-human', 'PPIL4-human', 'NOL12-human', 'EIF4G2-human',\n",
              "       'CPSF6-human', 'PTBP1-human', 'TBRG4-human', 'XRN2-human',\n",
              "       'EIF3G-human', 'CPEB4-human', 'GRSF1-human', 'DDX24-human',\n",
              "       'RPS11-human', 'SRSF9-human', 'FMR1-human', 'MTPAP-human',\n",
              "       'GEMIN5-human', 'PRPF4-human'], dtype=object)"
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# get all unique protein names - all of our classes\n",
        "targets = metadata['Experiment target'].unique()\n",
        "targets"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "This step takes about 50 minutes on a single core machine."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 49,
          "referenced_widgets": [
            "241b9837bc2a471f9fcad6520f2c7be6",
            "5cbdea56c4024497bbc52acb5103180f",
            "e0364314ef6a49f19901fd2b276369fe",
            "eeefa952e05e412a8d7bbd85923ea2dc",
            "1d2255b386c941f7b18a72f5aa8bd128",
            "66e5c8ea84404ed9a40ca380bd24c3a7",
            "02d3a5d8d91b4d3baa09a3161b71af41",
            "debd5dcebb004ba2b4c67f97adc660c7",
            "d8c6c7a7b528419583627cd125e0f66b",
            "e11ca5b8000e4427ad97480cba021b09",
            "7b8312b48b8b437d9bb014c14e5147ec"
          ]
        },
        "id": "TuB5J3_G3Llk",
        "outputId": "3601ea10-ed99-4262-d526-5e409d691eef"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/tmp/ipykernel_21605/2383095856.py:4: TqdmDeprecationWarning: This function will be removed in tqdm==5.0.0\n",
            "Please use `tqdm.notebook.tqdm` instead of `tqdm.tqdm_notebook`\n",
            "  for target in tqdm_notebook(targets):\n"
          ]
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "535259d6621a461c8b41da884cd5125b",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "  0%|          | 0/150 [00:00<?, ?it/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Problem with file  ENCFF100JBV.bed.gz . Trying again.\n"
          ]
        }
      ],
      "source": [
        "BED_HEADER = ['chrom', 'chromStart', 'chromEnd', 'name', 'score', 'strand', 'thickStart', 'thickEnd', 'itemRgb', 'blockCount', 'blockSizes', 'blockStarts']\n",
        "ALLOWED_CHROMOSOMES = ['chr' + str(i) for i in range(1, 22)] + ['chrX', 'chrY', 'chrMT']\n",
        "\n",
        "for target in tqdm_notebook(targets):\n",
        "\n",
        "  protein_name = target[: target.index('-')]\n",
        "  rows = metadata[metadata['Experiment target'] == target]\n",
        "\n",
        "  # In general, it is not a good practice to create a DataFrame in loop by appending. \n",
        "  # But since we know that we have only one or two rows for these targets, we can use it.\n",
        "  df = pd.DataFrame([], columns=['chr', 'start', 'end', 'strand'])\n",
        "  for index, row in rows.iterrows():\n",
        "\n",
        "    local_file = row['File accession'] + '.bed.gz'\n",
        "\n",
        "    # downloading files in while-try, because network connection might break sometimes\n",
        "    success = False\n",
        "    while not success:\n",
        "      try:\n",
        "        request.urlretrieve(row['File download URL'], local_file)\n",
        "        success = True\n",
        "      except:\n",
        "        print('Problem with file ', local_file, '. Trying again.')\n",
        "    reads = pd.read_csv(local_file, sep='\\t', compression='gzip', header=None, names=BED_HEADER)\n",
        "    \n",
        "    # keep just necessary columns\n",
        "    reads = reads[['chrom', 'chromStart', 'chromEnd', 'strand']]\n",
        "    # keep just sequences from chromosomes chr1 - ch22, chrX, chrX and chrMT\n",
        "    reads = reads[reads['chrom'].isin(ALLOWED_CHROMOSOMES)]\n",
        "    # removing outliers - keep just sequences >= 20 bp and <= 100 bp\n",
        "    reads = reads[((reads['chromEnd'] - reads['chromStart']) >= 20) & ((reads['chromEnd'] - reads['chromStart']) <= 100)]\n",
        "    # rename columns\n",
        "    reads.columns = ['chr', 'start', 'end', 'strand']\n",
        "\n",
        "    # add DataFrame from current file to DataFrame from previous file\n",
        "    df = pd.concat([df, reads], ignore_index=True)\n",
        "    \n",
        "    # cleaning - delete downloaded bed file\n",
        "    Path(local_file).unlink()\n",
        "\n",
        "  # add column with name of protein\n",
        "  df['protein_name'] = protein_name\n",
        "\n",
        "  filename = protein_name + '.csv.gz'\n",
        "  df.to_csv(BASE_FILE_PATH / filename, index=False, compression='gzip')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pG5ppwXZFLm2"
      },
      "source": [
        "## YAML file"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JndiLfH7FJpX",
        "outputId": "f93eb82f-3e79-4535-95f4-5ab93854bf6f"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'version': 0,\n",
              " 'classes': {'YBX3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SSB': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PABPN1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'UTP3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'WDR3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NIPBL': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'WDR43': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EXOSC5': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'UPF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'APOBEC3C': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'WRN': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PABPC4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FUS': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SUPV3L1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ZC3H8': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PHF6': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FXR2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FXR1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TROVE2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PCBP2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SUB1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SF3B1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'RBFOX2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'AGGF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SDAD1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPL': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPUL1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'KHDRBS1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FUBP3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'YWHAG': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FASTKD2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SND1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NONO': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TARDBP': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX52': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX51': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'STAU2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'BCLAF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HLTF': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'AQR': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPC': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'AKAP8L': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DHX30': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'AATF': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX6': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SBDS': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'LARP4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX55': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NSUN2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'U2AF2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'METAP2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'LSM11': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX21': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ABCF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SLTM': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SUGP2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TIAL1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SMNDC1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'BUD13': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'UCHL5': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SAFB2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NKRF': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EIF3D': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ZC3H11A': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SRSF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'XRCC6': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPK': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'RBM15': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'RPS3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SERBP1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'UTP18': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'GTF2F1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'GPKOW': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DGCR8': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'U2AF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PPIG': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPU': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PCBP1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TRA2A': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NPM1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'RBM5': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SF3B4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TAF15': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ZRANB2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX59': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PUS1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'LIN28B': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NCBP2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'POLR2G': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'GNL3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DKC1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PUM2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'QKI': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PUM1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'G3BP1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NIP7': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SAFB': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ZNF800': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EFTUD2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DROSHA': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EIF3H': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NOLC1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'BCCIP': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EWSR1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'IGF2BP1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'CSTF2T': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TIA1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FKBP4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'LARP7': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'AARS': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPM': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FTO': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SRSF7': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'MATR3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ILF3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FAM120A': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SLBP': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'RBM22': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX42': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'IGF2BP2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'HNRNPA1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'KHSRP': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'CSTF2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'XPO5': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'IGF2BP3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'ZNF622': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SF3A3': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'GRWD1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SFPQ': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'AKAP1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PRPF8': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'CDC40': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX3X': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PPIL4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'NOL12': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EIF4G2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'CPSF6': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PTBP1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'TBRG4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'XRN2': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'EIF3G': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'CPEB4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'GRSF1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'DDX24': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'RPS11': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'SRSF9': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'FMR1': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'MTPAP': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'GEMIN5': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'},\n",
              "  'PRPF4': {'type': 'fa.gz',\n",
              "   'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
              "   'extra_processing': 'ENSEMBL_HUMAN_GENOME'}}}"
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# YAML file with metadata\n",
        "# we store paths of reference fasta files\n",
        "\n",
        "desc = {\n",
        "  'version': 0,\n",
        "  'classes': {}\n",
        "}\n",
        "for target in targets:\n",
        "  name = target[: target.index('-')]\n",
        "  desc['classes'][name] = {\n",
        "      'type': 'fa.gz',\n",
        "      'url': 'http://ftp.ensembl.org/pub/release-97/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz',\n",
        "      'extra_processing': 'ENSEMBL_HUMAN_GENOME'\n",
        "  }\n",
        "\n",
        "with open(BASE_FILE_PATH / 'metadata.yaml', 'w') as fw:    \n",
        "    yaml.dump(desc, fw)\n",
        "\n",
        "desc"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "name": "create_datasets.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.8.10"
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "02d3a5d8d91b4d3baa09a3161b71af41": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1d2255b386c941f7b18a72f5aa8bd128": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_7b8312b48b8b437d9bb014c14e5147ec",
            "placeholder": "​",
            "style": "IPY_MODEL_e11ca5b8000e4427ad97480cba021b09",
            "value": " 150/150 [10:10&lt;00:00,  2.76s/it]"
          }
        },
        "241b9837bc2a471f9fcad6520f2c7be6": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_e0364314ef6a49f19901fd2b276369fe",
              "IPY_MODEL_eeefa952e05e412a8d7bbd85923ea2dc",
              "IPY_MODEL_1d2255b386c941f7b18a72f5aa8bd128"
            ],
            "layout": "IPY_MODEL_5cbdea56c4024497bbc52acb5103180f"
          }
        },
        "5cbdea56c4024497bbc52acb5103180f": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "66e5c8ea84404ed9a40ca380bd24c3a7": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "7b8312b48b8b437d9bb014c14e5147ec": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d8c6c7a7b528419583627cd125e0f66b": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "debd5dcebb004ba2b4c67f97adc660c7": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "e0364314ef6a49f19901fd2b276369fe": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_02d3a5d8d91b4d3baa09a3161b71af41",
            "placeholder": "​",
            "style": "IPY_MODEL_66e5c8ea84404ed9a40ca380bd24c3a7",
            "value": "100%"
          }
        },
        "e11ca5b8000e4427ad97480cba021b09": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "eeefa952e05e412a8d7bbd85923ea2dc": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_d8c6c7a7b528419583627cd125e0f66b",
            "max": 150,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_debd5dcebb004ba2b4c67f97adc660c7",
            "value": 150
          }
        }
      }
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
